### DISCUSSION TITLE
Connection to trotec machines

### DISCUSSION THREAD

**Discussion by Artemis-AD** [t-oster/VisiCut#651] [General]
Hi, 

Was their any success in connecting to a trotec machine? 

I have a few and can allow access to it remotely if it helps!

**Comment by t-oster** [comment]
Hi, as far as I know not. I once talked to some guy from trotec but we never got far. Do they have Ethernet or just USB?
If they have Ethernet you could try to execute a simple job like cutting a rectangle and capture the complete conversation with Wireshark, so we can get a clue what protocols they use.

**Comment by mgmax** [comment]
To give some background from my experience of writing a new laser driver: In general, every lasercutter uses its own specialized "language" and a good amount of reverse engineering (or getting documentation from the manufacturer) and programming is needed to add support for a new laser driver. In the past, all new laser drivers were developed by people who already had regular access to some laser device and wanted to make it work with VisiCut.  See also https://github.com/t-oster/VisiCut/wiki/Developing-a-new-Lasercutter-Driver .

With less time, you can at least capture the data (e.g. Windows "print to file" feature), have a look and compare with VisiCut's output for existing drivers. This can give a first estimate on how difficult the format is. Some comments claim that Trotec uses HPGL, and we already have a generic HPGL driver as well as drivers for the Epilog branded very specific HPGL dialect. So that may be a starting point for programming once you know which protocol it is.

**Comment by Artemis-AD** [comment]
I'll see what we can muster up here, I have only very basic programming experiance, but I know there would be a lot of very happy people if there was an option other then their own software.. I'll get cracking here and see what we can figure out, we've got 3 trotec machines, 2 of which have different programs, all connected via USB. 2 have internal computers which I'll need to get into to run the wireshark.
